src/timestamp9--1.0.1--1.1.0.sql

```diff
@@ -0,0 +1,3 @@
+CREATE FUNCTION timestamp9_nsnow() RETURNS timestamp9 AS
+'$libdir/timestamp9'
+LANGUAGE c IMMUTABLE STRICT PARALLEL SAFE LEAKPROOF;
```

Comment:
This function as it is implemented is not `IMMUTABLE`, but it's `VOLATILE`. Normally, the Postgres `now()/current_date` etc. functions are neither of these, but are `STABLE`, which means that they return the same value across different calls in the same statement. I think any 'now' function that timestamp9 provides should be STABLE too. This means it likely requires a different implementation though.

Reply:
I'm using two static variables to 'cache' the timestamp inside a transaction. It's a little bit different from PostgreSQL's `now()`, since `now()` returns the first statement's timestamp inside a transaction to avoid extra cost of calling `GetCurrentTimestamp()`[1]. In this this approach, `timestamp9_nsnow()` returns the timestamp of the first `timestamp9_nsnow()` function call.

[1] https://github.com/postgres/postgres/blob/a8a44828a2e15c95a15549063205b779f7055f1c/src/backend/access/transam/xact.c#L2111